src/tools/rest-backend/config.cpp

```diff
@@ -0,0 +1,152 @@
+/**
+ * @file
+ *
+ * @brief configuration helper functions
+ *
+ * @copyright BSD License (see doc/LICENSE.md or http://www.libelektra.org)
+ */
+
+#include <iostream>
+#include <string>
+
+#include <config.hpp>
+
+namespace kdbrest
+{
+
+/* set dummy variables for statics */
+
+// JWT settings
+std::string Config::jwt_encryption_key = std::string ();
+int Config::jwt_expiration_time = 0;
+
+// output settings
+std::string Config::output_default_entry_sort = std::string ();
+std::string Config::output_default_entry_sortby = std::string ();
+std::string Config::output_default_entry_filterby = std::string ();
+std::string Config::output_default_user_sort = std::string ();
+std::string Config::output_default_user_sortby = std::string ();
+std::string Config::output_default_user_filterby = std::string ();
+
+// permissions settings
+int Config::permissions_entry_create = 0;
+int Config::permissions_entry_edit = 0;
+int Config::permissions_entry_delete = 0;
+int Config::permissions_user_view = 0;
+int Config::permissions_user_edit = 0;
+int Config::permissions_user_delete = 0;
+int Config::permissions_default_rank = 0;
+
+// kdb settings
+std::string Config::kdb_path_configs = std::string ();
+std::string Config::kdb_path_users = std::string ();
+
+// api_specification settings
+std::string Config::api_specification_raw = std::string ();
+std::string Config::api_specification_html = std::string ();
+
+
+/**
+ * @brief initializes application configuration variables
+ * 
+ * can use compile-time defaults if no dynamic configuration
+ * is available. can also force the application to stop.
+ * 
+ * @param config the configuration to use
+ * @return 0 in case of success, 1 otherwise
+ */
+int Config::initializeConfiguration (cppcms::json::value & cfg)
+{
+	bool error = false;
+
+	// clang-format off
+	
+	// JWT settings
+	try {
+		Config::jwt_encryption_key = cfg.get<std::string> ("backend.jwt.encryption_key");
```

Comment:
Then I'll use the regex "[^a-zA-Z]" to sanitize the tool name. Renaming later should be possible, the cmake var `${tool}` should be used in every place where it matters.

For the profile stuff: I'll prepare it so that we could add it later and that "current" is the default for the moment.